src/scripts/main.js

```diff
@@ -1,3 +1,92 @@
 'use strict';
 
-// write code here
+const buttonContainer = document.querySelector('.container');
+const maxRow = 10;
+const minRow = 2;
+const maxColumn = 10;
+const minColumn = 2;
+
+buttonContainer.addEventListener('click', (e) => {
+  const target = e.target;
+  const tbody = document.querySelector('.field tbody');
+  const className = target.className.slice(0, -7);
+
+  switch (className) {
+    case 'append-row':
+      if (tbody.children.length < maxRow) {
+        const newRow = tbody.lastElementChild.cloneNode(true);
+
+        tbody.append(newRow);
+
+        if (tbody.children.length >= maxRow) {
+          target.setAttribute('disabled', true);
+        }
+
+        if (tbody.children.length > minRow) {
+          const removeRowButton = buttonContainer.querySelector('.remove-row');
+
+          removeRowButton.removeAttribute('disabled');
+        }
+      }
+      break;
+
+    case 'remove-row':
+      if (tbody.children.length > minRow) {
+        tbody.lastElementChild.remove();
+
+        if (tbody.children.length <= minRow) {
+          target.setAttribute('disabled', true);
+        }
+
+        if (tbody.children.length < maxRow) {
+          const addRowButton = buttonContainer.querySelector('.append-row');
+
+          addRowButton.removeAttribute('disabled');
+        }
+      }
+      break;
```

Comment:
you can avoid similar nesting if you'll check conditions in 'reverse direction'
```suggestion
    case 'remove-row':
      if (tbody.children.length < minRow) {
        break;
      }

      tbody.lastElementChild.remove();

      if (tbody.children.length <= minRow) {
        target.setAttribute('disabled', true);
      }

      if (tbody.children.length < maxRow) {
        addRowButton.removeAttribute('disabled');
      }

      break;
```

Reply:
corrected